lib/protobuf/dsl.ex

```diff
@@ -414,7 +414,7 @@ defmodule Protobuf.DSL do
   defp type_default(:sint32), do: 0
   defp type_default(:sint64), do: 0
   defp type_default(:bool), do: false
-  defp type_default({:enum, mod}), do: Code.ensure_loaded?(mod) && mod.key(0)
+  defp type_default({:enum, mod}), do: Code.ensure_compiled!(mod) && mod.key(0)
```

Comment:
If breaking changes aren't desirable, this one should fix both issues
```suggestion
  defp type_default({:enum, mod}) do
    if {:module, mod} == Code.ensure_compiled(mod), do: mod.key(0)
  end
```